modules/react/menu/lib/SubMenu.tsx

```diff
@@ -0,0 +1,246 @@
+import React from 'react';
+
+import {chevronRightSmallIcon} from '@workday/canvas-system-icons-web';
+
+import {
+  composeHooks,
+  createElemPropsHook,
+  createSubcomponent,
+  ExtractProps,
+  PropsWithModel,
+  subModelHook,
+  useForkRef,
+} from '@workday/canvas-kit-react/common';
+
+import {useMenuModel} from './useMenuModel';
+import {defaultMenuPopperOptions, useMenuPopper} from './MenuPopper';
+import {MenuItem, StyledMenuItem, useMenuItemArrowReturn, useMenuItemFocus} from './MenuItem';
+import {MenuCard} from './MenuCard';
+import {MenuList} from './MenuList';
+import {MenuDivider} from './MenuDivider';
+import {MenuOption} from './MenuOption';
+import {MenuGroup} from './MenuGroup';
+import {useListItemRegister, useListItemRovingFocus} from '@workday/canvas-kit-react/collection';
+import {Popper} from '@workday/canvas-kit-react/popup';
+
+type MenuModelConfig = (typeof useMenuModel)['TConfig'];
+
+export interface SubMenuProps
+  extends MenuModelConfig,
+    PropsWithModel<ReturnType<typeof useMenuModel>> {
+  /**
+   * The contents of the Menu. Can be `Menu` children or any valid elements.
+   */
+  children: React.ReactNode;
+}
+
+export const SubMenuPopper = createSubcomponent('div')({
+  modelHook: useMenuModel,
+  elemPropsHook: useMenuPopper,
+})<ExtractProps<typeof Popper>>(({children, ...elemProps}) => {
+  return (
+    <Popper placement="right-start" popperOptions={defaultMenuPopperOptions} {...elemProps}>
+      {children}
+    </Popper>
+  );
+});
+
+const useIntentTimer = (fn: Function, waitMs: number = 0): {start(): void; clear(): void} => {
+  const timer = React.useRef() as React.MutableRefObject<number | undefined>;
+
+  const start = () => {
+    timer.current = window.setTimeout(fn, waitMs);
+  };
+
+  const clear = () => {
+    window.clearTimeout(timer.current);
+    timer.current = undefined;
+  };
+
+  // be sure to clear our timeout
+  React.useEffect(() => {
+    return () => {
+      window.clearTimeout(timer.current);
+    };
+  }, [timer]);
+
+  return {
+    start,
+    clear,
+  };
+};
+
+export const useSubMenuTargetItem = composeHooks(
+  subModelHook(model => (model as any).UNSTABLE_parentModel!, useMenuItemFocus),
+  subModelHook(model => (model as any).UNSTABLE_parentModel!, useMenuItemArrowReturn),
+  subModelHook(model => (model as any).UNSTABLE_parentModel!, useListItemRovingFocus),
+  createElemPropsHook(useMenuModel)((model, ref) => {
+    const elementRef = useForkRef(ref, model.state.targetRef);
+    return {
+      ref: elementRef,
+    };
+  }),
+  subModelHook(model => (model as any).UNSTABLE_parentModel!, useListItemRegister),
+  createElemPropsHook(useMenuModel)(model => {
+    const currentTargetIdRef = React.useRef<string>();
+    const mouseEnterTimer = useIntentTimer(() => {
+      model.UNSTABLE_parentModel.events.goTo({id: currentTargetIdRef.current || ''});
+      model.events.show(event);
+    }, 300);
```

Comment:
Non blocker: Is this designs decision to have it be 300ms, should this be configurable?

Reply:
I don't know where this configuration will go. It seems like something that would go in the main `MenuModel` and cascade to every level. It may even be that certain `<Menu.Target>` should have a mouseenter + intent timer to open. We don't have this now, but that is common. For now, I think it is fine here. It can be added as a feature later.